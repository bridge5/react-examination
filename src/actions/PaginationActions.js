import * as types from '../constants/ActionTypes';

export function prevPage() {
    return {
        type: types.PREV_PAGE
    }
}

export function nextPage() {
    return {
        type: types.NEXT_PAGE
    }
}

export function goPage(pageNum) {
    return {
        type: types.GO_PAGE,
        pageNum
    }
}