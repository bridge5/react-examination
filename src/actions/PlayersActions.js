import * as types from '../constants/ActionTypes';

export function addPlayer(name, team, position) {
  return {
    type: types.ADD_PLAYER,
    name,
    team,
    position
  };
}

export function deletePlayer(id) {
  return {
    type: types.DELETE_PLAYER,
    id,
  };
}

export function starPlayer(id) {
  return {
    type: types.STAR_PLAYER,
    id,
  };
}
