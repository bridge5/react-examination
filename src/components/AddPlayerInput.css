:local(.addPlayerInput) {
  border-radius: 0;
  border-color: #abaaaa;
  border-left: 0;
  border-right: 0;
}
.addPlayerInputBox {
  display: flex;
  width: 500px;
}
