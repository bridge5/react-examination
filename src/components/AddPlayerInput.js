import React, { Component } from "react";
import classnames from "classnames";
import PropTypes from "prop-types";
import styles from "./AddPlayerInput.css";

class AddPlayerInput extends Component {
  render() {
    return (
      <div className="addPlayerInputBox">
        <input
          type="text"
          name="name"
          autoFocus={true}
          className={classnames("form-control", styles.addPlayerInput)}
          placeholder="Type the name of a player"
          value={this.state.name}
          onChange={this.handleChange.bind(this)}
          onKeyDown={this.handleSubmit.bind(this)}
        />
        <select 
          name="position"  
          value={this.state.position} 
          onChange={(e) => {
            this.handleChange(e);
          }}
          className={classnames("form-control", styles.addPlayerSelectInput)}
        >
        　<option value="SF">SF</option>
        　<option value="PG">PG</option>
        </select>
        <input 
          type="submit" 
          value="add"
          onClick={() => {
            this.handleSubmit.call(this, {which: 13})
          }}
        />
      </div>
    );
  }

  constructor(props, context) {
    super(props, context);
    this.state = {
      name: this.props.name || "",
      position: 'SF',
    };
  }

  handleChange(e) {
    const name = e.target.name;
    const value = e.target.value;
    this.setState({ [name]: value });
  }

  handleSubmit(e) {
    const name = this.state.name.trim();
    if (e.which === 13 && !!name) {
      this.props.addPlayer(name, this.state.position);
      this.setState({ name: "", position: 'SF' });
    }
  }
}

AddPlayerInput.propTypes = {
  addPlayer: PropTypes.func.isRequired
};

export default AddPlayerInput;
