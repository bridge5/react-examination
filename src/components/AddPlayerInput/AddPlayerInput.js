import React, { Component } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import styles from './AddPlayerInput.css';

class AddPlayerInput extends Component {
  render() {
    return (
      <input
        type="text"
        autoFocus={true}
        className={classnames('form-control', styles.addPlayerInput)}
        placeholder="Type the name of a player"
        value={this.state.name}
        onChange={this.handleChange.bind(this)}
        onKeyDown={this.handleSubmit.bind(this)}
      />
    );
  }

  constructor(props, context) {
    super(props, context);
    this.state = {
      name: this.props.name || '',
    };
  }

  handleChange(e) {
    this.setState({ name: e.target.value });
  }

  handleSubmit(e) {
    const name = e.target.value.trim();
    if (e.which === 13) {
      this.props.addPlayer(name);
      this.setState({ name: '' });
    }
  }
}

AddPlayerInput.propTypes = {
  addPlayer: PropTypes.func.isRequired,
};

export default AddPlayerInput;
