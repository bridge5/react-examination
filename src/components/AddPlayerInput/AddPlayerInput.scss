:local(.addPlayerInput) {
  border-radius: 0;
  border-color: #abaaaa;
  border-left: 0;
  border-right: 0;
  margin: 5px 0;
}
:local(.playPositionSelector) {
  margin: 5px 0;
}

:local(.wrap) {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  .btn {
    margin: 5px;
  }
}
