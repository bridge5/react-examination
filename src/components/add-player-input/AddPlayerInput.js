import React, { Component } from "react";
import classnames from "classnames";
import PropTypes from "prop-types";
import { POS_TYPES, Options } from "../../constants/posTypes";

import styles from "./AddPlayerInput.module.scss";

class AddPlayerInput extends Component {
  render() {
    return (
      <div className={styles.addPlayerFeild}>
        <input
          type="text"
          autoFocus={true}
          className={classnames("form-control", styles.addPlayerInput)}
          placeholder="Type the name of a player"
          value={this.state.name}
          onChange={this.handleChange.bind(this)}
          onKeyDown={this.handleSubmit.bind(this)}
        />
        <select onChange={this.handlePositionSelect}>{Options}</select>
      </div>
    );
  }

  constructor(props) {
    super(props);
    this.state = {
      name: "",
      position: POS_TYPES.SF,
    };
  }

  handleChange(e) {
    this.setState({ name: e.target.value });
  }
  handlePositionSelect = (e) => {
    this.setState({ position: e.target.value });
  };
  handleSubmit(e) {
    const name = e.target.value.trim();
    if (e.which === 13) {
      console.log(this.state.position);
      this.props.addPlayer({ name, position: this.state.position });
      this.setState({ name: "" });
    }
  }
}

AddPlayerInput.propTypes = {
  addPlayer: PropTypes.func.isRequired,
};

export default AddPlayerInput;
