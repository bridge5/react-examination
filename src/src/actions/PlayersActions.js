import * as types from '../constants/ActionTypes';

export function addPlayer(name) {
  return {
    type: types.ADD_PLAYER,
    name,
  };
}

export function deletePlayer(id) {
  return {
    type: types.DELETE_PLAYER,
    id,
  };
}

export function starPlayer(id) {
  return {
    type: types.STAR_PLAYER,
    id
  };
}

export function changePosition(data,id) {
  return {
    type: types.CHANGE_POSITION,
    data,
    id
  };
}
