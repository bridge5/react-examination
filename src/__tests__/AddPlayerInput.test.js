import React from 'react';
import AddPlayerInput from '../components/AddPlayerInput';
import renderer from 'react-test-renderer';
import TestUtils from 'react-dom/test-utils';
import ReactDOM from 'react-dom';
import { addPlayer } from "../actions/PlayersActions";

const player = [
  {
    name: 'LeBron James',
    team: 'LOS ANGELES LAKERS',
    position: 'SF',
    starred: true,
  },
  {
    name: 'Kevin Duran',
    team: 'GOLDEN STATE WARRIORS',
    position: 'SF',
    starred: false,
  },
  {
    name: 'Anthony Davis',
    team: 'NEW ORLEANS PELICANS',
    position: 'PF',
    starred: false,
  },
  {
    name: 'Stephen Curry',
    team: 'GOLDEN STATE WARRIORS',
    position: 'PG',
    starred: false,
  },
  {
    name: 'James Harden',
    team: 'HOUSTON ROCKETS',
    position: 'SG',
    starred: false,
  },
  {
    name: 'Kawhi Leonard',
    team: 'TORONTO RAPTORS',
    position: 'SF',
    starred: false,
  },
]

describe('AddPlayerInput', function() { 
  var AppElement = TestUtils.renderIntoDocument(<AddPlayerInput addPlayer={addPlayer}></AddPlayerInput>); 
  var DomElement = ReactDOM.findDOMNode(AppElement); 
  var input = DomElement.getElementsByTagName('input')[0]; 
  var submitBtn = DomElement.getElementsByTagName('input')[1]; 

  it('renders correctly', () => {
    const tree = renderer
      .create(<AddPlayerInput addPlayer={addPlayer}></AddPlayerInput>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('type should be text', function() { 
   expect(input.getAttribute('type')).toEqual('text'); 
  }); 

  it('should render 2 input', () => {
    const testRenderer = renderer.create(<AddPlayerInput addPlayer={addPlayer} />);
    const testInstance = testRenderer.root;
    expect(testInstance.findAll(node => node.type === 'input')).toHaveLength(2);
  });

  it('should empty name and set default position to SF', function() {
    TestUtils.Simulate.click(submitBtn);
    expect(AppElement.state).toEqual({
        name: "",
        position: "SF",
    });
  });
}); 
