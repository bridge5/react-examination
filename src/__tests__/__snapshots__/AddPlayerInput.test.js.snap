// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPlayerInput renders correctly 1`] = `
<div
  className="addPlayerInputBox"
>
  <input
    autoFocus={true}
    className="form-control"
    name="name"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Type the name of a player"
    type="text"
    value=""
  />
  <select
    className="form-control"
    name="position"
    onChange={[Function]}
    value="SF"
  >
    　
    <option
      value="SF"
    >
      SF
    </option>
    　
    <option
      value="PG"
    >
      PG
    </option>
  </select>
  <input
    onClick={[Function]}
    type="submit"
    value="add"
  />
</div>
`;
